### ISSUE TITLE
Problem after setting up

### ISSUE THREAD

**Issue by Xentran** [EvoTM/EvoSC#54]
After setting up EvoSC it fails to boot/start.
(i'm using the trackmania 2020 server.)

**To Reproduce**
After i start it via CMD this error shows up.

**Screenshots/Logs**

> [14:33:16] EscRun->execute(): Starting...
[14:33:16] Database::init(): Connecting to database...
[14:33:16] Database::init(): Database connected.

> PHP Fatal error:  Uncaught TypeError: Argument 1 passed to EvoSC\Classes\RestClient::init() must be of the type string, null given, called in C:\Users\Administrator\Desktop\EvoSC\core\Commands\EscRun.php on line 150 and defined in C:\Users\Administrator\Desktop\EvoSC\core\Classes\RestClient.php:37
> Stack trace:
> #0 C:\Users\Administrator\Desktop\EvoSC\core\Commands\EscRun.php(150): EvoSC\Classes\RestClient::init(NULL)
> #1 C:\Users\Administrator\Desktop\EvoSC\vendor\symfony\console\Command\Command.php(255): EvoSC\Commands\EscRun->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
> #2 C:\Users\Administrator\Desktop\EvoSC\vendor\symfony\console\Application.php(1000): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
> #3 C:\Users\Administrator\Desktop\EvoSC\vendor\symfony\console\Application.php(271): Symfony\Component\Console\Application->doRunCommand(Object(EvoSC\Comma in C:\Users\Administrator\Desktop\EvoSC\core\Classes\RestClient.php on line 37
> 
> Fatal error: Uncaught TypeError: Argument 1 passed to EvoSC\Classes\RestClient::init() must be of the type string, null given, called in C:\Users\Administrator\Desktop\EvoSC\core\Commands\EscRun.php on line 150 and defined in C:\Users\Administrator\Desktop\EvoSC\core\Classes\RestClient.php:37
> Stack trace:
> #0 C:\Users\Administrator\Desktop\EvoSC\core\Commands\EscRun.php(150): EvoSC\Classes\RestClient::init(NULL)
> #1 C:\Users\Administrator\Desktop\EvoSC\vendor\symfony\console\Command\Command.php(255): EvoSC\Commands\EscRun->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
> #2 C:\Users\Administrator\Desktop\EvoSC\vendor\symfony\console\Application.php(1000): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
> #3 C:\Users\Administrator\Desktop\EvoSC\vendor\symfony\console\Application.php(271): Symfony\Component\Console\Application->doRunCommand(Object(EvoSC\Comma in C:\Users\Administrator\Desktop\EvoSC\core\Classes\RestClient.php on line 37

**EvoSC (please complete the following information):**
 - EvoSC-Version: [0.91.0] 

Did i miss something?

Thanks for any help or comment!

**Comment by brakerb** [comment]
EvoSC was unable to load your server-name. Try again with the latest update.

**Comment by Xentran** [comment]
Thanks for your reply.

I just downloaded it today from github. 
ManiaServer is also up2date.

I downloaded EvoSC via Github client. Is there maybe a problem ?

**Comment by brakerb** [comment]
It does not run well on Windows, currently only linux is fully supported.
But I can see at one line there's a wrong path, I can check that and also that your database-tables probably do not exist. Try running `php esc migrate` and then run it again to see if it helps.

**Comment by mantazzo** [comment]
I've been trying to get it working on Windows too, and been progressing forward. The "php esc migrate" command throws an "FilePathNotAbsoluteException" in File.php Line 237. After analyzing what it looks for, I understood the issue - the substr here searches for the directory separator, and on Windows, directory path doesn't start with it (starts with your usual C:, etc.) - for now, commented out the whole part and it seems migrate is now working; I changed another line too, just in case:
`return dirname(realpath(__DIR__)) . str_replace('/', DIRECTORY_SEPARATOR, '/cache/' . $filename);`
Seems like the server wanted to start, but it threw me some other issues now, which I'm analyzing.

**Comment by brakerb** [comment]
I will check the absolute path thing.
About the cURL error/warning: you can ignore that

**Comment by brakerb** [comment]
FilePathNotAbsoluteException should not occurr on windows anymore

**Comment by mantazzo** [comment]
Seems improved, but after trying to start a fresh install, I once again got that Exception.

After looking a bit, I found out it happens in MapController.php, line 53 - for some reason the makeDir command fails. I tried to check the command, and it seems it's correct, however, after changing the preg_match there (all I did was removing the NOT sign) it worked.
Can't understand why it worked like that - as I understand, the regex there should return true to get the false (to make the full argument incorrect), and after checking with online regex tools it should work like that, but it seems like for some reason that regex generates false, which throws an error. Not sure what the reason is for that, but removing the ! sign seemed to be enough for start.
